### DISCUSSION TITLE
Layered tasks?

### DISCUSSION THREAD

**Discussion by nxy7** [casey/just#1774] [Ideas]
Hello, I'm using just for some time already and wondered if there's better way of managing my project than I am currently.
I'm using just to manage my monorepo which includes frontend, backend and some related jobs (like launching k8s locally). Right now I have root level task with something like
```
# backend related commands
backend subcommand="":
	@just backend/{{subcommand}}
```

and in backend/Justfile
```
_default:
	@just --list

test:
	cargo nextest r

...
```
You get my point. This way i can run `just backend test`, it separates commands nicely into multiple domains (frontend, backend, utilities), but sadly it's not 'natively' supported and I think this workaround is not that nice. First - it pollutes project structure with Justfiles. Commands I'm running are not very long and I'd prefer one long file. Secondly help message displays `backend subcommand=""` so it leaves blank subcommand which I don't particularly love.
Is there any better way to accomplish what I'm trying to do? Is it out of the scope of the project? Ideally I'd like to avoid things like `just backend-test` as I'd like to display list of tasks as a tree that's easier to navigate for newcomers.

**Comment by casey** [comment]
The long term plan to support this is #383, but at the moment, there isn't a great way to do this.

**Reply by nxy7** [reply]
Okay, knowing that it's on the radar is more than enough for me. I was looking for issues/discussions related to this topic, but I guess didn't word it correctly and didn't find this issue so thanks for the info :-)
